docs/spec_draft.md

```diff
@@ -663,6 +664,142 @@ produces an implementation-defined value.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.dot_general
+
+### Semantics
+
+Performs general dot product between `lhs` and `rhs` over dimensions specified
+in lhs/rhs batching and contracting dimensions and produces a `result` tensor.
+
+This general form allows dot product between multi-dimensional tensors, and to
+contract multiple dimensions at a time. It also allows to specify the batching
+dimensions, which determines the dimensions to avoid summing during contraciton.
+
+More formally, `result[id, ik] = sum(lhs[md, mk] * rhs[nd, nk])` where:
+
+1. `id == md == nd` where `md == lhs_batching_dimensions[j]` and `nd == rhs_batching_dimensions[j]` for `j` in [0, size(`lhs_batching_dimensions`))
+2. `id = md` where `d` $\notin$ `lhs_contracting_dimensions` $\cup$ `lhs_batching_dimensions`
+3. `id = nd` where `d` $\notin$ `rhs_contracting_dimensions` $\cup$ `rhs_batching_dimensions`
+
+The summation is from `0` to prod(dim(`lhs_contracting_dimensions`, `i`)) for all `i` in [0, size(`lhs_contracting_dimensions`))
+for all `mk`, `nk` combinations defined as:
+* `mk` in [0, dim(`lhs`, `k`)) for all `k` $\in$ `lhs_contracting_dimensions`
+* `nk` in [0, dim(`rhs`, `k`)) for all `k` $\in$ `rhs_contracting_dimensions`
+
+`precision_config` controls the tradeoff between speed and accuracy for
+computations on accelerator backends. This can be one of the following:
+
+  * `DEFAULT`: Fastest mode, but least accurate.
+  * `HIGH`: Slower, but more accurate.
+  * `HIGHEST`: Slowest, but most accurate.
+
+### Inputs
+
+| Name                         | Type                                         |
+|------------------------------|----------------------------------------------|
+| `lhs`                        | tensor of any supported types                |
+| `rhs`                        | tensor of any supported types                |
+| `lhs_batching_dimensions`    | 1-dimensional tensor constant of type `si64` |
+| `rhs_batching_dimensions`    | 1-dimensional tensor constant of type `si64` |
+| `lhs_contracting_dimensions` | 1-dimensional tensor constant of type `si64` |
+| `rhs_contracting_dimensions` | 1-dimensional tensor constant of type `si64` |
+| `precision_config`           | constant of type `si64`                      |
```

Comment:
How about 

constant of type `enum {DEFAULT, HIGHEST, HIGH}` ?

@burmako @ghpvnist 
IMO introducing enum does not need introduction. We wanted to convey that precision_config takes special constant values w/o dealing with how the enum case is [actually stored](https://github.com/llvm/llvm-project/blob/1f54641d33d57d85d925e72ffda9c76aef4e9601/mlir/include/mlir/IR/EnumAttr.td#L42).

Reply:
From https://github.com/openxla/stablehlo/blob/07578f02f7a6d1477e43a3c8f52db6c5ec33eb68/stablehlo/dialect/StablehloEnums.td#L46 it seems like an optional array, where we can have multiple constraints specified. Do you know what is the semantics if the the array.

In the constraints we can mention that the `precision config` is optional and what the default value is.

Reply:
Fair point. I have no objections. I'll make the changes once everyone is in agreement.

Reply:
Looking back at previous PRs with the new enum phrasing, I'll change this accordingly.